airflow/providers/google/cloud/operators/cloud_storage_transfer_service.py

```diff
@@ -246,7 +254,17 @@ def execute(self, context: 'Context') -> dict:
             gcp_conn_id=self.gcp_conn_id,
             impersonation_chain=self.google_impersonation_chain,
         )
-        return hook.create_transfer_job(body=self.body)
+        result = hook.create_transfer_job(body=self.body)
+        if NAME in result:
+            self.xcom_push(context, key="name", value=result[NAME])
```

Comment:
this is a part of the `result` so we don't need to push it also to cross comunication, whole `result` is available there